src/http.rs

```diff
@@ -77,9 +80,11 @@ pub async fn do_http_request(
     }
     if !is_authorized(&caller, Auth::FreeRpc) {
         let cycles_available = ic_cdk::api::call::msg_cycles_available128();
-        if cycles_available < cycles_cost {
+        let cycles_cost_with_collateral =
+            cycles_cost + COLLATERAL_CYCLES_PER_NODE * get_nodes_in_subnet() as u128;
+        if cycles_available < cycles_cost_with_collateral {
```

Comment:
Is there a reason why `request_cost` is not called directly instead of re-doing the same computation `cycles_cost + COLLATERAL_CYCLES_PER_NODE * get_nodes_in_subnet() as u128`?

Reply:
I refactored usages of this calculation into a shared `get_cost_with_collateral()` function (since `request_cost()` is a canister method). Is this essentially what you were imagining?